shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXConnection.scala

```diff
@@ -146,10 +192,36 @@ class UCXConnection(peerExecutorId: Int, ucx: UCX) extends Connection with Loggi
     composeTag(composeUpperBits(peerClientId, bufferMsgType), bufferTag)
   }
 
-  private def composeTag(upperBits: Long, lowerBits: Long): Long = {
+  protected def composeRequestAmId(requestType: RequestType.Value): Int = {
+    val amId = requestType.id
+    if ((amId & 0x0000000F) != amId) {
+      throw new IllegalArgumentException(
+        s"Invalid request amId, it must be 4 bits: ${TransportUtils.toHex(amId)}")
+    }
+    amId
+  }
+
+  protected def composeRequestHeader(executorId: Long, txId: Long): Long = {
+    composeTag(executorId << 32, txId)
+  }
+
+  def extractExecutorId(header: Long): Long = {
+    (header & 0xFFFFFFFF00000000L) >> 32
```

Comment:
shift then mask, bc if header has the MSB set then I'll get sign extension on the shift.

Write some unit tests for this?